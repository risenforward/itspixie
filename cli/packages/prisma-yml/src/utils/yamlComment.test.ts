import { replaceYamlValue } from './yamlComment'

describe('replaceYamlValue', () => {
  test('when document is clean', () => {
    const input = `\
    endpoint: https://eu1.prisma.sh/public-asdf/my-service/dev
    datamodel: datamodel.graphql`

    expect(
      replaceYamlValue(input, 'endpoint', 'http://localhost:4466'),
    ).toMatchSnapshot()
  })

  test('when comments already exist', () => {
    const input = `\
    #anothercomment: asdasd
    endpoint: https://eu1.prisma.sh/public-asdf/my-service/dev

    #endpoint: asdasd
    datamodel: datamodel.graphql`

    expect(
      replaceYamlValue(input, 'endpoint', 'http://localhost:4466'),
    ).toMatchSnapshot()
  })

  test('when key does not yet exist', () => {
    const input = `\
    datamodel: datamodel.graphql`

    expect(
      replaceYamlValue(input, 'endpoint', 'http://localhost:4466'),
    ).toMatchSnapshot()
  })
})
