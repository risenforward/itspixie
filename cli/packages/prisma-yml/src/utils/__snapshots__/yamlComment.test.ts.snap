// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`replaceYamlValue when comments already exist 1`] = `
"    #anothercomment: asdasd
    #endpoint: https://eu1.prisma.sh/public-asdf/my-service/dev
endpoint: http://localhost:4466
    #endpoint: asdasd
    datamodel: datamodel.graphql"
`;

exports[`replaceYamlValue when document is clean 1`] = `
"    #endpoint: https://eu1.prisma.sh/public-asdf/my-service/dev
endpoint: http://localhost:4466    datamodel: datamodel.graphql"
`;

exports[`replaceYamlValue when key does not yet exist 1`] = `
"    datamodel: datamodel.graphql
endpoint: http://localhost:4466"
`;
